docs/design/descheduler.md

```diff
@@ -0,0 +1,102 @@
+* Rescheduling pods based on real resource utilization instead of request resource.
+* Support custom configured rescheduling strategies.
+
+## Design
+### WorkFlow
+1. Filter Resources which are selected to be evicted potentially according to the filter chain, for example, queue filter
+and label filter.
+2. Get through the chain of rescheduling strategies and filter resources which are to be evicted.
+3. Evict the Pods attached to these resources.
+4. Execute the process above periodically.
+
+### Resource Filter
+* Queue Filter
+
+  `Queue filtering` will filter resources in specified queue. Then the rescheduling process only works on the result set.
+
+* Label Filter
+
+  `Label Filter` will filter pods with specified labels. Then the rescheduling process only works on the result set.
+
+### Rescheduling Strategy
+* OfflineOnly
+
+    `OfflineOnly`, abbreviated as `OLO`, will only pick out offline workloads, which are attached with annotation
+`preemptable: true`, and then reschedule them only.
+
+* LowPriorityFirst
+
+    `LowPriorityFirst`, abbreviated as `LPF`, will sort workloads by priority and then reschedule pods in ascending 
+order.
+
+* ShortLifeTimeFirst
+
+  `ShortLifeTimeFirst`, abbreviated as `SLTF`, will sort workloads by running time. Pods with the shortest lift time 
+will be rescheduled first. This strategy can make sure workloads with long task type goes healthily without interruptions.
+
+* BigObjectFirst
+
+    `BigObjectFirst`, abbreviated as `BOF`, will select workloads which request the most **dominate resource** and reschedule
+them first. It helps improve system throughout and avoid small workloads' starvation.
+
+* MoreReplicasFirst
+  
+    `MoreReplicasFirst`, abbreviated as `MRF`, will sort workloads by replicas number. Workloads with the most replicas
+will be rescheduled first. This strategy is friendly to `gang scheduling` for it will consider the `minAvailable` in 
+volcano jobs.
+
+### Metrics
+All the decisions made by rescheduling strategies will consider the metrics from `Prometheus`. Namely, Volcano will 
+list the real node resource utilization and pod distribution instead of request resource. Basically, usage of`CPU` and 
+`Memroy` will be collected. Other resource such as `GPU` can be extended.
+
+## Implementation
+* Load and parse user configurations about rescheduling.
+```yaml
+actions: "enqueue, allocate, backfill, rescheduling"  ## add 'rescheduling' at the end of the actions
+tiers:
+  - plugins:
+      - name: priority
+      - name: gang
+      - name: conformance
+      - name: strategy       ## enable rescheduling strategies 
+        arguments:
+          offlineOnly: true
+          lowPriorityFirst: true
+  - plugins:
+      - name: overcommit
+      - name: drf
+      - name: predicates
+      - name: proportion
+      - name: nodeorder
+      - name: binpack
+rescheduling:    ## configurations for rescheduling
```

Comment:
it should be a plugin, instead of importing all code from descheduler.

Reply:
Not import all code from descheduler, reimplement part of its functions.